csrc/scheduler/normalization_inner_outer.cpp

```diff
@@ -238,80 +240,108 @@ PersistentBufferStorageParams getPersistentBufferStorageParams(
   int64_t available_regs = scheduler_utils::register_file_size_56k;
   buffer_params.smem_overhead = smem_overhead;
 
-  // (1) init the buffer_params by putting all the persistent tensors in
-  // registers
-  buffer_params.regs_buffer_size = total_buffer_size;
-  buffer_params.smem_buffer_size = 0;
-
-  // (2) If the available register is larger than current register buffer size,
-  // no need to move buffers to shared memory, return early.
-  if (buffer_params.regs_buffer_size <= available_regs) {
+  // (1) Try to put all the buffers into registers.
+  if (total_buffer_size <= available_regs) {
+    buffer_params.regs_buffer_size = total_buffer_size;
+    buffer_params.smem_buffer_size = 0;
     buffer_params.has_enough_regs_and_smem = true;
     return buffer_params;
   }
```

Comment:
This `if` branch will be removed in #3221. It should be removed with a slight change of the heuristics to avoid large regressions.